### PR TITLE
sql: change SHOW ZONE CONFIGURATION to use FROM instead of FOR

### PR DESCRIPTION
**Pull request by andyyang890**
The SHOW ZONE CONFIGURATION statements were changed to use
FROM instead of FOR to standardize the use of FROM with SHOW.

Refs: #57638

Release note (sql change): This patch overrides the release note
added in #58740. The SHOW ZONE CONFIGURATION statement has been
changed to use FROM instead of FOR.

### CODE REVIEW

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -5137,9 +5137,13 @@ show_zone_stmt:
     $$.val = &tree.ShowZoneConfig{}
   }
 
-for_or_from:
+for_implicit_alias_from:
```
Comment (otan) [inline pkg/sql/parser/sql.y:5140]: maybe `for_with_implicit_from_alias` which is more descriptive when reading the BNF diagrams

Comment (andyyang890) [reply pkg/sql/parser/sql.y]: Changed.

File: pkg/sql/parser/sql.y [hunk 2]
```diff
@@ -5137,9 +5137,13 @@ show_zone_stmt:
     $$.val = &tree.ShowZoneConfig{}
   }
 
-for_or_from:
+for_with_implicit_from_alias:
   FOR
-| FROM
+| FROM { /* SKIP DOC */ }
+
+from_with_implicit_for_alias:
```
Comment (otan) [inline pkg/sql/parser/sql.y:5144]: this isn't used yet right? in that case we should probably skip it

**Review by otan — APPROVED** [review]

File: pkg/sql/parser/sql.y (hunk 2 above)
Comment (andyyang890) [reply pkg/sql/parser/sql.y:5144]: Oh I'm planning on using it in my next PR fixing the rest of the FOR/FROM aliases. Is it ok if I keep it in this commit?

Comment (otan) [reply pkg/sql/parser/sql.y:5144]: oh, we're going both ways? for which examples?

Comment (otan) [reply pkg/sql/parser/sql.y:5144]: (in this case i'd rather introduce it in the same commit!)

Comment (andyyang890) [reply pkg/sql/parser/sql.y:5144]: Oh I thought that's what we wanted to do from #57638 but I'll double check.

Comment (otan) [reply pkg/sql/parser/sql.y:5144]: I don't think so.

But you can merge this by deleting those lines ;) !

Comment (andyyang890) [reply pkg/sql/parser/sql.y:5140]: FYI based on the discussion in #57638, the statement will be changed to have FROM be the documented version but with FOR retained as an implicit alias.

**Review by otan — APPROVED** [review]

**Comment by andyyang890** [comment]
bors r=otan

**Pull request merged**